container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/engine_functions/create_engine.go

```diff
@@ -239,13 +240,34 @@ func CreateEngine(
 		usedPorts[debugServerDockerPort] = docker_manager.NewManualPublishingSpec(uint16(engineDebugServerPort))
 	}
 
+	// Configure GitHub Auth by writing the provided token to a volume that's accessible by the engine
+	githubAuthStorageVolObjAttrs, err := objAttrsProvider.ForGitHubAuthStorageVolume()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred retrieving object attributes for GitHub auth storage.")
+	}
+	githubAuthStorageVolNameStr := githubAuthStorageVolObjAttrs.GetName().GetString()
+	githubAuthStorageVolLabelStrs := map[string]string{}
+	for labelKey, labelValue := range githubAuthStorageVolObjAttrs.GetLabels() {
+		githubAuthStorageVolLabelStrs[labelKey.GetString()] = labelValue.GetString()
+	}
+	// This volume is created idempotently (like logs storage volume) and just write the token to the file everytime the engine starts
+	if err = dockerManager.CreateVolume(ctx, githubAuthStorageVolNameStr, githubAuthStorageVolLabelStrs); err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred creating GitHub auth storage volume.")
+	}
+	githubAuthStorageCreator := github_auth_storage_creator.NewGitHubAuthStorageCreator(gitAuthToken)
```

Comment:
`githubAuthStorageCreator` writes whatever the github token passed to `backend.CreateEngine` to a volume called `kurtosis-github-auth-storage`. This volume is then mounted to `APIC`s so they have access to GitHub auth. 

Initially, `githubAuthToken` was passed as an env var to the `engine` and `APIC`, but that would've leaked the token during `kurtosis dump` or `kurtosis enclave dump`, so I went with the mounted volume approach for providing the engine and APIC with the auth token. It's considerably more code but safer.